src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -626,6 +625,56 @@ public byte[] Http2AltSvcOriginUri
             return await GetHttpConnectionAsync(request, cancellationToken).ConfigureAwait(false);
         }
 
+        private Http2Connection? GetHttp2ConnectionAcceptingNewStreams()
+        {
+            IReadOnlyList<Http2Connection>? localConnections = _http2Connections;
+
+            if (localConnections == null)
+            {
+                return null;
+            }
+
+            for (int i = 0; i < localConnections.Count; i++)
+            {
+                Http2Connection http2Connection = localConnections[i];
+
+                if (!_poolManager.Settings.EnableMultipleHttp2Connections || http2Connection.CanAddNewStream)
+                {
+                    TimeSpan pooledConnectionLifetime = _poolManager.Settings._pooledConnectionLifetime;
+                    if (http2Connection.LifetimeExpired(Environment.TickCount64, pooledConnectionLifetime))
+                    {
+                        // Connection expired.
+                        http2Connection.Dispose();
+                        InvalidateHttp2Connection(http2Connection);
+                    }
+                    else
+                    {
+                        return http2Connection;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private void AddHttp2Connection(Http2Connection newConnection)
+        {
+            lock (SyncObj)
+            {
+                Http2Connection[]? localHttp2Connections = _http2Connections;
```

Comment:
nit: Do you need localHttp2Connections variable while holding SyncObj? You shouldn't need to worry about someone else modifying the array.

Reply:
Same as [above](https://github.com/dotnet/runtime/pull/38748#discussion_r452794909).